src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceTestTransformer.java

```diff
@@ -401,11 +423,35 @@ private void addUnitTestImports(InterfaceContext context) {
     typeTable.saveNicknameFor("org.junit.BeforeClass");
     typeTable.saveNicknameFor("org.junit.Test");
     if (context.getInterfaceConfig().hasPageStreamingMethods()) {
-      typeTable.saveNicknameFor("com.google.api.gax.core.PagedListResponse");
+      typeTable.saveNicknameFor("com.google.api.gax.paging.PagedListResponse");
     }
     if (context.getInterfaceConfig().hasLongRunningOperations()) {
       typeTable.saveNicknameFor("com.google.protobuf.Any");
     }
+    switch (context.getProductConfig().getTransportProtocol()) {
```

Comment:
Not a big deal, but consider using https://google.github.io/guava/releases/19.0/api/docs/com/google/common/collect/ImmutableMultimap.html .

That way you can declare all the imports elsewhere, then
```java
for (String type : multimap.get(context.getProductConfig().getTransportProtocol())) {
  typeTable.saveNicknameFor(type);
}
```

Similar strategy can be used below as well.

Reply:
I don't quite follow. When does the multimap get populated? I like having all the imports listed in the `add.*Imports()` functions, with each function corresponding to a snippet file, which usually defines which types will be used.

Reply:
Ah, I usually populate the multimap in a `private static final` variable. If you like it this way better, I don't feel strongly against it.